docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md

```diff
@@ -9,25 +9,247 @@ menu:
+- `SET default_transaction_read_only = TRUE` applies the read-only setting to all statements and transaction blocks that follow.
+- `/*+ Set(transaction_read_only true) */ SELECT` applies only to the current transaction/select statement.
+
+## Examples
+
+This example uses follower reads since the **transaction** is marked read-only.
+
+```sql
+set yb_read_from_followers = true;
+start transaction read only;
+SELECT * from t WHERE k='k1';
+commit;
+```
+
+```output
+ k  | v
+----+----
+ k1 | v1
+```
+
+This example uses follower reads since the **session** is marked read only.
+
+```sql
+set session characteristics as transaction read only;
+set yb_read_from_followers = true;
+SELECT * from t WHERE k='k1';
+```
+
+```output
+ k  | v
+----+----
+ k1 | v1
+(1 row)
+```
+
+The following examples use follower reads since **pg_hint** can be used during PREPARE, CREATE FUNCTION, and SELECT to do follower reads.
+
+{{< note title="Note" >}}
+The pg_hint needs to be applied at the prepare/function-definition stage and not at the `execute` stage.
+{{< /note >}}
+
+```sql
+set session characteristics as transaction read write;
+set yb_read_from_followers = true;
+
+PREPARE select_stmt(text) AS
+/*+ Set(transaction_read_only on) */
+SELECT * from t WHERE k=$1;
+
+EXECUTE select_stmt(‘k1’);
+```
+
+```output
+ k  | v
+----+----
+ k1 | v1
+(1 row)
+```
+
+```sql
+set session characteristics as transaction read write;
+set yb_read_from_followers = true;
+
+CREATE FUNCTION func() RETURNS text AS
+$$ /*+ Set(transaction_read_only on) */
+SELECT * from t WHERE k=1 $$ LANGUAGE SQL;
+CREATE FUNCTION
+SELECT func();
+```
+
+```output
+ k  | v
+----+----
+ k1 | v1
+(1 row)
+```
+
+```sql
+set session characteristics as transaction read write;
+set yb_read_from_followers = true;
+/*+ Set(transaction_read_only on) */
+SELECT * from t WHERE k='k1';
+```
+
+```output
+----+----
+ k1 | v1
+(1 row)
+```
+
+A **join** example that uses follower reads.
+
+```sql
+create table table1(k int primary key, v int);
+create table table2(k int primary key, v int);
+insert into table1 values (1, 2), (2, 4), (3, 6), (4,8);
+insert into table2 values (1, 3), (2, 6), (3, 9), (4,12);
+set yb_read_from_followers = true;
+set session characteristics as transaction read only;
+select * from table1, table2 where table1.k = 3 and table2.v = table3.v;
+```
+
+```output
+ k | v | k | v
+---+---+---+---
+ 3 | 6 | 2 | 6
+(1 row)
+```
+
+The following examples demonstrate **staleness** after enabling the `yb_follower_read_staleness_ms` property.
+
+```sql
+set session characteristics as transaction read write;
+select * from t where k = 'k1';
+```
+
+```output
+ k  | v
+----+----
+ k1 | v1
+(1 row)
+```
+
+```sql
+UPDATE t SET  v = 'v1+1' where k = 'k1';
+/* sleep 10s */
+
+UPDATE t SET  v = 'v1+2' where k = 'k1';
+/* sleep 10s */
+
+select * from t where k = 'k1';
+```
+
+```output
```

Comment:
Might be worth adding some commentary along with..

For example, here I would say:

This selects the latest version of the row because the transaction setting for the session is `read write`.